src/Google/Cache/File.php

```diff
@@ -151,11 +151,14 @@ private function getCacheDir($file, $forWrite)
     // use the first 2 characters of the hash as a directory prefix
     // this should prevent slowdowns due to huge directory listings
     // and thus give some basic amount of scalability
-    $dirHash = substr(md5($file), 0, 2);
+    $fileHash = substr(md5($file), 0, 2);
+    $userHash = md5(get_current_user());
+    $dirHash = $fileHash . DIRECTORY_SEPARATOR . $userHash;
```

Comment:
Ok, I wanted to make sure the collisions are not the problem and I found the actual collision is between binary values way longer than the usual usernames.
```
4dc968ff0ee35c209572d4777b721587d36fa7b21bdc56b74a3dc0783e7b9518afbfa200a8284bf36e8e4b55b35f427593d849676da0d1555d8360fb5f07fea2
```
collides with
```
4dc968ff0ee35c209572d4777b721587d36fa7b21bdc56b74a3dc0783e7b9518afbfa202a8284bf36e8e4b55b35f427593d849676da0d1d55d8360fb5f07fea2
```

Thanks for being patient